cmd/thanos/sidecar.go

```diff
@@ -298,13 +338,31 @@ func runSidecar(
 	return nil
 }
 
+func validatePrometheus(ctx context.Context, logger log.Logger, promURL *url.URL, tsdbPath string) error {
+	flags, err := promclient.ConfiguredFlags(ctx, logger, promURL)
+	if err != nil {
+		return errors.Wrap(err, "configured flags; failed to check flags")
+	}
+	// Check if min-block-time and max-block-time are the same.
+	if flags.TSDBMinTime != flags.TSDBMaxTime {
+		return errors.New("TSDB Min-block-time mismatches with Max-block-time")
+	}
+	// Check if block-time equals 2h.
+	if flags.TSDBMinTime != model.Duration(2*time.Hour) {
+		level.Warn(logger).Log("msg", "TSDB Max-block-time and Min-block-time should be configured to 2h", "block-time", flags.TSDBMinTime)
```

Comment:
Shouldn't we return an error here and forbid the user from continuing on if it doesn't match 2 hours? We probably need to be strict here